src/core/queries.ml

```diff
@@ -98,14 +112,23 @@ let handle_query : Sig_state.t -> proof_state option -> p_query -> unit =
       let h = get_pp_hint sym in
       if h <> Unqual then out 1 " [%a]" pp_hint h;
       (match !(sym.sym_def) with
-      | Some t -> out 1 " ≔ %a\n" pp_term t
+      | Some t ->
+          out 1 " ≔ %a\n" pp_term t;
+          Some (Format.asprintf " ≔ %a" pp_term t)
       | None ->
           out 1 "\n";
-          List.iter (fun r -> out 1 "%a\n" pp_rule (sym, r)) !(sym.sym_rules))
+          let rule oc r = Format.fprintf oc "%a\n" pp_rule (sym, r) in
+          out 1 "%a" (List.pp rule "") !(sym.sym_rules);
+          Some (
+            List.fold_left (^) ""
+              (List.map (fun r -> Format.asprintf "%a\n" pp_rule (sym, r))
+                !(sym.sym_rules))))
```

Comment:
May be simplified as `Some (Format.asprintf "%a" (List.pp rule "") !(sym.sym_rules))`.